R/verb-joins.R

```diff
@@ -614,10 +614,7 @@ join_two_table_alias <- function(names, from) {
   if (from[1] != "as" && from[2] != "as") {
     # self join of named table
     if (!is.na(names[1]) && !is.na(names[2]) && identical(names[1], names[2])) {
-      out <- c(
```

Comment:
I think I decided to use this pattern because it is slightly more expressive than the generic LHS and RHS.
If we want to keep this alternative possibilities would be:

* abbreviate the table names e.g. by using `abbreviate()` (though I'm not quite convinced about its results...)
* use `substr()` to shorten the string and then append LHS and RHS

Reply:
Thank you for the context and suggestions; I agree. I went with `abbreviate()` based on it seeing like the clearest path forward. Hopefully the options I specified add a bit of robustness, while recognizing we don't expect this case to be hit very often.